### PR TITLE
Dev ship data

### PR DESCRIPTION
**Pull request by kok-foong**
Requested by you-know-who to display more ship data that are relevant to regulations.

Main addition:
1) New derivation that is derived from ship and derived using ShipDataAgent.
2) ShipDataAgent
    - supposed to calculate new quantities based on some available ship data. At the moment just adds static values for energy efficiency, specific fuel consumption, carbon intensity index, specific CO2 emission
    - Design intention is to hopefully have some simple models to calculate additional quantities from readily available data
    - Currently limited by the amount of data available, need information that ship owners don't typically reveal, such as power rate, type of fuel used, specific fuel consumption

### CODE REVIEW

File: JPS_Ontology/ontology/ontodispersion/OntoDispersion.owl
Comment (ChungTingLao) [inline JPS_Ontology/ontology/ontodispersion/OntoDispersion.owl:1]: Version should be incremented.

File: JPS_VIRTUALSENSOR/ShipDataAgent/pom.xml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    <groupId>com.cmclinnovations</groupId>
+    <artifactId>ShipDataAgent</artifactId>
+    <version>1.0-dev-ship-data-SNAPSHOT</version>
+    <packaging>war</packaging>
+    
+    <!-- Parent POM -->
+    <parent>
+        <groupId>uk.ac.cam.cares.jps</groupId>
+        <artifactId>jps-parent-pom</artifactId>
+        <version>2.2.0</version>
```
Comment (ChungTingLao) [inline JPS_VIRTUALSENSOR/ShipDataAgent/pom.xml:16]: Should use latest parent pom 2.3.0

File: JPS_VIRTUALSENSOR/ShipDataAgent/pom.xml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                <artifactId>maven-dependency-plugin</artifactId>
+			</plugin>
+		</plugins>
+    </build>
+
+	<dependencies>
+        <dependency>
+			<groupId>uk.ac.cam.cares.jps</groupId>
+			<artifactId>jps-base-lib</artifactId>
+			<version>1.40.1</version>
```
Comment (ChungTingLao) [inline JPS_VIRTUALSENSOR/ShipDataAgent/pom.xml:85]: Should use latest base lib 1.43.2

File: JPS_VIRTUALSENSOR/ShipDataAgent/pom.xml [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-web</artifactId>
+            <scope>runtime</scope>
+        </dependency>
+		
+		<!-- for interacting with the stack -->
+        <dependency>
+            <groupId>com.cmclinnovations</groupId>
+            <artifactId>stack-clients</artifactId>
+            <version>1.22.0</version>
```
Comment (ChungTingLao) [inline JPS_VIRTUALSENSOR/ShipDataAgent/pom.xml:129]: Should use latest stack client 1.28.0

**Comment by ChungTingLao** [comment]
Checked locally that it works for static JSON and CSV ship data. Can be merged once the comments are addressed.

File: JPS_Ontology/ontology/ontodispersion/OntoDispersion.owl [hunk 4]
```diff
@@ -317,7 +327,7 @@
         <rdfs:subClassOf rdf:resource="https://www.theworldavatar.com/kg/ontodispersion/ShipStaticProperty"/>
         <rdfs:comment>Physical dimension of the ship</rdfs:comment>
         <rdfs:isDefinedBy rdf:resource="https://www.theworldavatar.com/kg/ontodispersion/"/>
-        <rdfs:label>Dimension</rdfs:label>
+        <rdfs:label>Dimension [m]</rdfs:label>
```
[the remark this answers is not in the document]
Comment (kok-foong) [reply JPS_Ontology/ontology/ontodispersion/OntoDispersion.owl:330]: It's AIS dimensions, the literal is a JSON with 4 values, a, b, c ,d. https://www.researchgate.net/figure/Reference-point-for-AIS-position-and-overall-dimensions-of-ship-Note-that-dimension-B-on_fig5_273123985

**Review by ChungTingLao — APPROVED** [review]

**Pull request merged**